modules/nw-udn-cr.adoc

```diff
@@ -39,27 +39,72 @@ spec:
 <3> This field specifies the topology configuration, it can be `layer2` or `layer3`.
 <4> Specifies `Primary` or `Secondary`. `Primary` is the only `role` specification supported in {product-version}.
 <5> The maximum transmission units (MTU) is an optional field. The default value is `1400`. The boundary for IPv4 is 574, and for IPv6 it is 1280.
-<6> Specifies the subnet to be used for the network across the cluster. Supports both IPv6 and dual-stack. For example, `192.168.100.0/24`,`2001:DBB::/64`. Dual-stack may set two subnets otherwise only one is allowed. When the `topology` field is set to `Layer3`, the subnet is split into smaller subnets for every node. Accepted format for subnets when `Layer3` is set are: `172.16.0.0/16/24`. For `Layer2` values in `topology` field, standard CIDR ranges are accepted. If omitted the network only provides `Layer2` communication and you must configure IP addresses.
-<7> Specifies the `subnets` used inside the OVN-Kubernetes network topology. If omitted, the OVN-Kubernetes network plugin assigns one, which is subject to change over time.
+<6> For `Layer2` topology types the following specifies config details for the `subnet` field:
++
+* The subnets field is optional.
+
+* The subnets field is of type `string` and accepts standard CIDR formats for both IPv4 and IPv6.
+
+* The subnets field accepts one or two items. For two items, they must be of a different family. For example, subnets values of `10.100.0.0/16` and `2001:db8::/64`.
+
+* Layer2 subnets may be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+
+* The Layer2 `subnets` field is mandatory when `ipamLifecycle` field is specified.
+<7> It is not recommended to set this field without explicit need and understanding of the OVN network topology. Dual-stack clusters may set 2 subnets, one for each IP family, otherwise only 1 subnet is allowed. This field is only allowed for `Primary` network. When omitted, the platform will choose a reasonable default which is subject to change over time.
 <8> Specifies the IP address management system (IPAM). This field is optional and allowed when `topology` is `layer2`. The
 `subnets` field must be specified when this field is specified. The `Persistent` value specifies that workloads have persistent IP addresses. Assigned by the container network interface (CNI) and used by OVN-Kubernetes to program pod IP addresses. You must not change this for pod annotations.
 
+.. Create a YAML file, such as `my-layer-three-udn.yaml`, to define your request with content as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: UserDefinedNetwork
+metadata:
+  name: udn-2 # <1>
+  namespace: <some_custom_namespace>
+spec:
+  topology: Layer3 # <2>
+  layer3: <3>
+    role: Primary # <4>
+    mtu: 9000 # <5>
+    subnets:
+      - "10.0.0.0/24"
+      - "2001:db8::/64" # <6>
```

Comment:
shouldn't we provide an example w/ hostSubnet defined ?

Reply:
done

Reply:
Nice; maybe also good to define one subnet per family, for dual-stack scenarios.

Reply:
From the code, I believe only /64 IPv6 hostsubnet  are supported and we should explicitly mention that cc @ormergi @maiqueb

Reply:
Updated note 6 to include `/64` limitation. @ormergi and @maiqueb

Reply:
OK, just found this in the code:
https://github.com/ovn-org/ovn-kubernetes/blob/369ac907f9123cfaff68868cad3389d20201508c/go-controller/pkg/config/utils.go#L96

Does seem we only support /64 subnets. It deserves a callout for sure.

Reply:
Thanks @maiqueb That was an easy find. Not sure why I couldn't locate. @JoeAldinger that limitation is just for hostsubnet so callout can be changed to `For IPv6, only `/64` host subnet length is supported`

Reply:
Updated, I removed the note from Layer2 because it doesn't seem relevant, but updated it for `hostsubnet` in Layer3 using IPv6. @anuragthehatter and @maiqueb